### PR TITLE
Added Redux Persist

### PR DESCRIPTION
**Pull request by NIveX2169**
- [x] Fix the Signup Auth redirection Bug.

- [x] Persist User-Auth state

### CODE REVIEW

**Review by pooranjoyb — COMMENTED** [review]
Great job, @NIveX2169! 🎉 Thank you for submitting your pull request. Your contribution is valuable and we appreciate your efforts to improve our project.<p>We will promptly review your changes and offer feedback. Keep up the excellent work! Kindly remember to check our [contributing guidelines](https://github.com/pooranjoyb/popShop/blob/dev/CONTRIBUTING.md)</p>

File: package.json
Comment (pooranjoyb) [inline package.json:1]: Take a pull from `dev`. I guess I changed the `@rollup` back to `optionalDependencies` coz someone in Windows faced an issue for that.

**Review by pooranjoyb — CHANGES_REQUESTED** [review]

File: package.json
Comment (NIveX2169) [reply package.json:1]: sure ill do it what i have observed right now is that i have not pushed toast.. so ill do that along.

**Comment by pooranjoyb** [comment]
Hey @NIveX2169 donot remove the `optionalDependency`, I said just to remove the `devDependency`

File: package.json [hunk 1]
```diff
@@ -36,12 +35,16 @@
...
     "daisyui": "^4.7.2",
+    "@rollup/rollup-linux-x64-gnu": "^4.17.2",
     "eslint": "^8.56.0",
     "eslint-plugin-react-hooks": "^4.6.0",
     "eslint-plugin-react-refresh": "^0.4.5",
     "postcss": "^8.4.38",
     "typescript": "^5.2.2",
     "vite": "^5.1.0"
+  },
+  "optionalDependencies": {
```
Comment (pooranjoyb) [inline package.json:46]: rollup should be here 😞

**Review by pooranjoyb — CHANGES_REQUESTED** [review]

**Review by pooranjoyb — APPROVED** [review]

**Review by pooranjoyb — APPROVED** [review]

**Pull request merged**